s3/pkg/meta/db/drivers/tidb/bucket.go

```diff
@@ -309,38 +331,67 @@ func (t *TidbClient) ListObjects(ctx context.Context, bucketName, marker, verIdM
 					continue
 				}
 			}
+
 			var o *Object
-			Strver := strconv.FormatUint(version, 10)
-			o, err = t.GetObject(ctx, bucketname, name, Strver)
+			strVer := strconv.FormatUint(version, 10)
+			o, err = t.GetObject(ctx, bname, name, strVer)
 			if err != nil {
-				err = handleDBError(err)
+				log.Errorf("err:%v\n", err)
 				return
 			}
+
 			count += 1
 			if count == maxKeys {
 				nextMarker = name
 			}
-			if count == 0 {
-				continue
-			}
+
 			if count > maxKeys {
 				truncated = true
 				exit = true
 				break
 			}
+
 			retObjects = append(retObjects, o)
 		}
-		if loopcount == 0 {
+
+		tend := time.Now()
+		tdur = tend.Sub(tqueryend).Nanoseconds()
+		if tdur/1000000 > TimeDur {
+			log.Debugf("slow get when list objects, time:%d\n", tdur)
+		}
+
+		if loopcount < MaxObjectList {
 			exit = true
 		}
 		if exit {
 			break
 		}
 	}
 	prefixes = helper.Keys(commonPrefixes)
+
 	return
 }
 
+func (t *TidbClient) CountObjects(ctx context.Context, bucketName, prefix string) (*utils.ObjsCountInfo, error) {
+	var sqltext string
+	rsp := utils.ObjsCountInfo{}
+	var err error
+	if prefix == "" {
+		sqltext = "select count(*),sum(size) from objects where bucketname=?;"
+		err = t.Client.QueryRow(sqltext, bucketName).Scan(&rsp.Count, &rsp.Size)
+	} else {
+		filt := prefix + "%"
+		sqltext = "select count(*),sum(size) from objects where bucketname=? and name like ?;"
```

Comment:
SQL injection possible here...please check. What if, i pass in prefix as 'delete * from objects'

Reply:
Hi @rhsakarpos , we did not construct the SQL statement ourselves, but use the argument placeholders, in this way, I think,  the underlying driver for database/sql will ultimately be aware of what special characters it needs to handle and will escape them for us.

Reply:
exactly...in order NOT to leave things to the underlying database driver implementation (which is most of the time opaque), we should handle any possible injection in our code